### PR TITLE
Add CI and testing suite

### PR DESCRIPTION
**Pull request by jcohenadad**
The purpose of this PR is to:
- establish a continuous integration (CI) framework using Github actions;
- establish a test suite with [MOxUnit](https://github.com/MOxUnit/MOxUnit);
- upload mock data on newly-created [OSF repository](https://osf.io/ntb32/)

## Local testing

Below is a docker solution to mimik the online GH environment

<details>

```bash
docker pull ubuntu:16.04
docker run -it ubuntu:latest
# while inside docker
apt-get -y -qq update
apt-get -y install wget unzip emacs git
apt-get -y install octave liboctave-dev octave-pkg-dev
# install testing suite
git clone https://github.com/MOxUnit/MOxUnit.git
cd MOxUnit/
make install
# install repos
git clone https://github.com/shimming-toolbox/shimming-toolbox.git
cd shimming-toolbox/
git checkout -b jca/ci-github-action origin/jca/ci-github-action
# download mock data
wget -O data.zip https://osf.io/7d2j5/download
unzip data.zip 
```

</details>

### CODE REVIEW

**Comment by kousu** [comment]
Wow https://github.com/joergbrech/moxunit-action/ is great! Unfortunately there's a lot of compatibility problems with the current code base and Octave that will need to be addressed first before we can run it in CI. You can see that Action uses Octave here

https://github.com/joergbrech/moxunit-action/blob/f7682ab17e6b552d39059d5d5c68fd2d176e67fc/entrypoint.sh#L113

and it's never going to not because Matlab needs a license to run which Github isn't going to pay for just for us.

So I am very much in support of going in this direction but we probably need to do a separate Octave-compatibility PR for it. Perhaps this PR could just add the test data -- and make sure the files in example/ work reproducibly on Matlab, the next PR could fix Octave compatibility, and the last one could integrate CI.

**Comment by kousu** [comment]
Also I would recommend rebase-squashing all the commits to `.github/workflows/main.yml` together. There's no way to debug CI without uploading to CI, but once it's working there's nothing to be learned from keeping the experimental stage around.

**Comment by jcohenadad** [comment]
> Also I would recommend rebase-squashing all the commits to .github/workflows/main.yml together. There's no way to debug CI without uploading to CI, but once it's working there's nothing to be learned from keeping the experimental stage around.

i know it's ugly... unfortunately i have some other emergencies to take care off and i won,t have the time to do a proper cleaning

**Comment by po09i** [comment]
I squashed the commits to the best of my ability in this [branch](https://github.com/shimming-toolbox/shimming-toolbox/tree/ad/ci-github-actions-squashed). If this work, I can apply the changes to this branch.

**Comment by jcohenadad** [comment]
@po09i could you do the squash directly in my branch and then leave your comments— thanks

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        cd tests
+        wget -O data.zip https://osf.io/7d2j5/download
+        unzip data.zip
+        rm data.zip
+        cd ..
+
+    - name: Run MOxUnit tests
+      uses: joergbrech/moxunit-action@master
+      with:
+        test: tests
```
Comment (po09i) [inline .github/workflows/main.yml:29]: The argument name should be called "tests" according to [mox-unit-action repo](https://github.com/joergbrech/moxunit-action/tree/f7682ab17e6b552d39059d5d5c68fd2d176e67fc). This should remove the warning that was thrown when running the step: run MoxUnit tests.

Comment (po09i) [inline .github/workflows/main.yml]: The argument name should be called "tests" according to [mox-unit-action repo](https://github.com/joergbrech/moxunit-action/tree/f7682ab17e6b552d39059d5d5c68fd2d176e67fc). This should remove the warning that was thrown when running the step: run MoxUnit tests.

**Review by po09i — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (jcohenadad) [reply .github/workflows/main.yml]: good point! would you mind fixing it directly in the branch? thx

**Review by po09i — APPROVED** [review]

**Pull request merged**